apis/v1beta1/config.go

```diff
@@ -318,6 +323,10 @@ func (c *Config) GetExporterPorts(logger logr.Logger) ([]corev1.ServicePort, err
 	return c.getPortsForComponentKinds(logger, KindExporter)
 }
 
+func (c *Config) GetExtensionPorts(logger logr.Logger) ([]corev1.ServicePort, error) {
+	return c.getPortsForComponentKinds(logger, KindExtension)
+}
+
 func (c *Config) GetAllPorts(logger logr.Logger) ([]corev1.ServicePort, error) {
 	return c.getPortsForComponentKinds(logger, KindReceiver, KindExporter)
```

Comment:
You should add extensions here as well. Then you'll automatically get the ports added to the container spec.

Reply:
The idea is good but I am already exposing `extension` ports in the new service. Just out of curiosity, will this have any advantage? 🤔

Reply:
Added.